daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/InterpreterTest.scala

```diff
@@ -22,13 +24,16 @@ class InterpreterTest extends WordSpec with Matchers with TableDrivenPropertyChe
 
   private implicit def id(s: String): Ref.Name = Name.assertFromString(s)
 
+  private val noPackages: PureCompiledPackages = PureCompiledPackages(Map.empty).right.get
+
   private def runExpr(e: Expr): SValue = {
-    val machine = Speedy.Machine.fromExpr(
-      expr = e,
-      compiledPackages = PureCompiledPackages(Map.empty).right.get,
-      scenario = false,
+    val machine = Speedy.Machine(
+      compiledPackages = noPackages,
       submissionTime = Time.Timestamp.now(),
       initialSeeding = InitialSeeding.NoSeed,
+      noPackages.compiler.unsafeCompile(e),
+      Set.empty,
+      Set.empty
```

Comment:
Named arguments for the last three args as well?

Reply:
Should we have a convenience function akin to `Machine.buildForScenario` for the pattern with `Timestamp.now()`,`NoSeed` and two times `Set.empty` as well? This appears a few times.